ui/src/app/utils/subscribable.tsx

```diff
@@ -0,0 +1,43 @@
+import * as fp from 'lodash/fp';
+
+export interface Subscribable<T> {
+  subscribe: (Function) => { unsubscribe: () => void};
+  next: (newValue: T, oldValue: T) => void;
+}
+
+export interface Atom<T> {
+  get: () => T;
+  set: (value: T) => void;
+  subscribe: Subscribable<T>['subscribe'];
+  reset: () => void;
+}
+
+export function subscribable<T>(): Subscribable<T> {
+  let subscribers = [];
+  return {
+    subscribe: fn => {
+      subscribers = fp.concat(subscribers, [fn]);
+      return {
+        unsubscribe: () => {
+          subscribers = fp.without([fn], subscribers);
+        }
+      };
+    },
+    next: (...args) => {
```

Comment:
I'm surprised the type system is OK with this. Should this have the more constrained method signature found in the interface?

Reply:
added typing